pkg/csi/service/wcpguest/controller_helper.go

```diff
@@ -160,23 +161,32 @@ func checkForSupervisorPVCCondition(ctx context.Context, client clientset.Interf
 		if !ok {
 			continue
 		}
-		if checkPVCCondition(ctx, pvc, reqCondition) {
+		if checkPVCCondition(ctx, pvc, reqCondition, reqSize) {
 			return nil
 		}
 	}
-	return fmt.Errorf("supervisor persistentVolumeClaim %s in namespace %s not in %q condition within %d seconds",
-		pvcName, ns, reqCondition, timeoutSeconds)
+	return fmt.Errorf("supervisor PersistentVolumeClaim %s in namespace %s not in %s condition and %s size "+
+		"within %d seconds", pvcName, ns, reqCondition, reqSize.String(), timeoutSeconds)
 }
 
 func checkPVCCondition(ctx context.Context, pvc *v1.PersistentVolumeClaim,
-	reqCondition v1.PersistentVolumeClaimConditionType) bool {
+	reqCondition v1.PersistentVolumeClaimConditionType, reqSize *resource.Quantity) bool {
 	log := logger.GetLogger(ctx)
 	for _, condition := range pvc.Status.Conditions {
-		log.Debugf("PersistentVolumeClaim %s in namespace %s is in %s condition", pvc.Name, pvc.Namespace, condition.Type)
+		log.Debugf("PersistentVolumeClaim %s in namespace %s is in %s condition", pvc.Name,
+			pvc.Namespace, condition.Type)
 		if condition.Type == reqCondition {
```

Comment:
what is the `reqCondition` you are expecting here?

Reply:
We are checking for "FileSystemResizePending" condition on PVC. I have added it in method doc as well now.